Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -81,55 +135,89 @@ Quality RawDataReaderCheck::check(std::map<std::string, std::shared_ptr<MonitorO
           vGoodBC.push_back(i);
         }
       }
-      if (vBadBC.size() > 0)
+      if (vBadBC.size() > 0) {
         result = Quality::Bad;
-      else if (vMediumBC.size() > 0)
+      } else if (vMediumBC.size() > 0) {
         result = Quality::Medium;
-      else
+      } else {
         result = Quality::Good;
-    }
-    if (mo->getName() == "inputs" || mo->getName() == "classes") {
+      }
+    } else {
       auto* h = dynamic_cast<TH1F*>(mo->getObject());
```

Comment:
Are you certain that anything else you might get will be a `TH1F*` and can be treated with this part of the code?

Also, please check if `h` is not a nullptr (which could be when the cast has failed) before you access it, so your code does not crash in such a case, but produces a valid error message instead.

Reply:
yes, we are producing 5 histograms in the task. One needs to be treated separately and the other 4 can be handled with the code below "else"
The check for h being a null pointer will be added.